### PR TITLE
Fix animation bugs on Module Library

### PR DESCRIPTION
**Pull request by Whelpley**

### CODE REVIEW

File: assets/js/ModuleCategory.js [hunk 1]
```diff
@@ -76,44 +50,32 @@ export default class ModuleCategory extends React.Component {
...
+      : <div className='first-level' onMouseEnter={this.toggleCollapse} >
+          <div className='cat-container closed-ML-cat' >
+            <span className={icon}></span>
+          </div>
+        </div>
 
-    // 'on-*' props trigger an objection from Facebook, see: https://reactjs.org/warnings/unknown-prop.html
-    var collapse = (this.props.libraryOpen)
-      ? <Collapse className='' isOpen={isOpen}>
+    var moduleList = (this.props.libraryOpen)
```
Comment (jstray) [inline assets/js/ModuleCategory.js]: Generally I prefer if statements to ?:, except for very short and simple expressions.

File: assets/js/ModuleCategory.js [hunk 2]
```diff
@@ -40,27 +35,6 @@ export default class ModuleCategory extends React.Component {
     this.props.setOpenCategory(newCollapsed ? null : this.props.name); // tell parent, so it can close other cats
   }
 
-  onEntering() {
```
Comment (jstray) [inline assets/js/ModuleCategory.js]: Deleting code feels so good! Less code, less bugs.

File: assets/js/ModuleCategory.test.js [hunk 3]
```diff
@@ -55,18 +53,30 @@ describe('ModuleCategory ', () => {
...
   
-    // it('Clicking on a category will expand it', () => { 
-    //   expect(true).toBe(true);
-    // });
-
-    // it('Expanding one category will collapse the others', () => { 
-    //   expect(true).toBe(true);
-    // });
+    it('Click events on a category will toggle its module list display', () => { 
+      expect(true).toBe(true);
```
Comment (jstray) [inline assets/js/ModuleCategory.test.js]: Uh, why?

File: assets/js/ModuleCategory.test.js [hunk 4]
```diff
@@ -55,18 +53,30 @@ describe('ModuleCategory ', () => {
...
+      // access isOpen property of Collapse, check that it is closed
+      expect(moduleList.get(0).props.isOpen).toBe(false);
+      // simulate a click on category
+      category.simulate('click');
+      expect(wrapper).toMatchSnapshot();      
+      // check isOpen, should be open
+      expect(moduleList.get(0).props.isOpen).toBe(true);
+      // another click
+      category.simulate('click');      
+      // isOpen should be closed
```
Comment (jstray) [inline assets/js/ModuleCategory.test.js]: Add another snapshot here to check that it actually closed?

File: assets/js/ModuleCategory.test.js (hunk 3 above)
Comment (Whelpley) [reply assets/js/ModuleCategory.test.js]: No reason, deleting this.

File: assets/js/ModuleCategory.js (hunk 1 above)
Comment (Whelpley) [reply assets/js/ModuleCategory.js]: Noted, changing

**Pull request merged**